python_modules/dagster/dagster/_core/snap/snap_to_yaml.py

```diff
@@ -0,0 +1,44 @@
+import json
+from typing import Any, Optional
+
+from dagster._config.snap import ConfigSchemaSnapshot, ConfigTypeSnap
+from dagster._utils.yaml_utils import dump_run_config_yaml
+
+
+def _safe_json_loads(json_str: Optional[str]) -> object:
+    try:
+        return json.loads(json_str) if json_str else None
+    except json.JSONDecodeError:
+        return None
+
+
+def default_values_yaml_from_type_snap(
+    snapshot: ConfigSchemaSnapshot,
+    type_snap: ConfigTypeSnap,
+) -> str:
+    return dump_run_config_yaml(default_values_from_type_snap(type_snap, snapshot))
+
+
+def default_values_from_type_snap(type_snap: ConfigTypeSnap, snapshot: ConfigSchemaSnapshot) -> Any:
+    defaults_by_field = {}
+
+    for field_name in type_snap.field_names:
+        field = type_snap.get_field(field_name)
+
+        # no default placeholder
+        default_value = ...
+
+        default_value_as_json = field.default_value_as_json_str
+        if default_value_as_json:
+            default_value = _safe_json_loads(default_value_as_json)
+        if not default_value_as_json and snapshot:
```

Comment:
looks like this could just be an else since `snapshot` always has a value according to the type signature